src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ExtensionComponent.scala

```diff
@@ -0,0 +1,70 @@
+package org.broadinstitute.dsde.workbench.leonardo.db
+
+import org.broadinstitute.dsde.workbench.leonardo.model.UserJupyterExtensionConfig
+
+case class ExtensionRecord(clusterId: Long, extensionType: String, name: String, value: String)
+
+trait ExtensionComponent extends LeoComponent {
+  this: ClusterComponent =>
+
+  import profile.api._
+
+  class ExtensionTable(tag: Tag) extends Table[ExtensionRecord](tag, "NOTEBOOK_EXTENSIONS") {
+    def clusterId = column[Long]("clusterId")
+
+    def extensionType = column[String]("extensionType", O.Length(254))
+
+    def name = column[String]("name", O.Length(254))
+
+    def path = column[String]("path", O.Length(254))
+
+    def cluster = foreignKey("FK_CLUSTER_ID", clusterId, clusterQuery)(_.id)
+
+    def uniqueKey = index("IDX_LABEL_UNIQUE", (clusterId, extensionType, name), unique = true)
+
+    def * = (clusterId, extensionType, name, path) <> (ExtensionRecord.tupled, ExtensionRecord.unapply)
+  }
+
+  object extensionQuery extends TableQuery(new ExtensionTable(_)) {
+
+    def save(clusterId: Long, extensionType: String, name: String, value: String): DBIO[Int] = {
+      extensionQuery += ExtensionRecord(clusterId, extensionType, name, value)
+    }
+
+    def saveAllForCluster(clusterId: Long, userJupyterExtensionConfig: Option[UserJupyterExtensionConfig]) = {
+      userJupyterExtensionConfig match {
+        case Some(ext) => extensionQuery ++= marshallExtensions(clusterId, ext)
+        case None => DBIO.successful(0)
+      }
+    }
+
+    def getAllForCluster(clusterId: Long): DBIO[UserJupyterExtensionConfig] = {
+      extensionQuery.filter {
+        _.clusterId === clusterId
+      }.result map { recs =>
+        val nbExtensions = (recs.filter(_.extensionType == "nbExtension") map { rec => rec.name -> rec.value }).toMap
```

Comment:
Maybe make the extension types constants (or better yet, enums) in LeonardoModel instead of hardcoded strings here.